Userland/Libraries/LibPartition/EBRPartitionTable.cpp

```diff
@@ -28,16 +24,51 @@ ErrorOr<NonnullOwnPtr<EBRPartitionTable>> EBRPartitionTable::try_to_initialize(N
     return table;
 }
 
+class EBRPartitionTable::ClonableDeviceRef {
+public:
+#ifdef KERNEL
+    using BackingType = Kernel::StorageDevice&;
+#else
+    using BackingType = NonnullOwnPtr<Core::File>;
+#endif
```

Comment:
Do we have a chance to implement `SeekableStream` for `StorageDevice`? If so, we should probably consider this a temporary solution at best, and aim for making everything run using `MaybeOwned<SeekableStream>` instead.

Reply:
See below: I suggest a new class for the common API, since there is so litle overlap. I named it `PartitionableDevice`.